src/redux/actions/sync.js

```diff
@@ -82,45 +51,96 @@ export function doSetDefaultAccount(success, failure) {
   };
 }
 
-export function doGetSync(password) {
+export function doSetSync(oldHash, newHash, data, password) {
```

Comment:
Should we rename this to something more descriptive? Right now, this calls `/sync/set`, and then sets the default account if the hashes are different, and finally unlocks the the wallet if it's locked. It is also possible for the `account_unlock` call to fail with an error. If that happens, should `SET_SYNC_FAILED` be dispatched?

Alternatively, we could have two methods. One for calling `/sync/set` and the another one calling `/sync/set` with the extra steps.

Reply:
I think it makes it a lot nicer to work with if theres just one method. But I can add a parameter to the function that triggers the additional steps

Reply:
After looking at this again after your comment I realized it should have been pulled out. I put it in `doGetSync` with a flag that starts the additional steps. This shouldn't require any changes from you.